File: leetcode/jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int max_index = 0;
        for (int i = 0; i < nums.size() - 1; i++)
        {
            if (nums[i] == 0 && i >= max_index)
            {
                return false;
            }
            max_index = max(max_index, i + nums[i]);
        }
        return true;
    }
};
// 4 2 0 2 1 0 3 3 0 11 2 0 2
// 

File: leetcode/jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        max_idx = 0
        for i in range(len(nums) - 1):
            if nums[i] == 0 and max_idx == i:
                return False
            max_idx = max(max_idx, i + nums[i])
        
        return True